File: Backend/src/main/java/com/sparfuchs/DTO/PurchaseDTO.java
package com.sparfuchs.DTO;

import java.time.LocalDateTime;
import java.util.List;

public record PurchaseDTO(Long purchaseId,
                          Long storeId,
                          LocalDateTime lastUpdated,
                          List<PurchaseProductResponseDTO> products,
                          boolean isCompleted ,
                          double totalSpent,
                          double totalSaved) {
}

File: Frontend/SparfuchsApp/app/src/main/java/com/example/sparfuchsapp/data/remote/dto/PurchaseDTO.kt
package com.example.sparfuchsapp.data.remote.dto

import com.squareup.moshi.JsonClass
import java.time.LocalDateTime

@JsonClass(generateAdapter = true)
data class PurchaseDTO (
    val purchaseId: Long,
    val storeId: Long,
    val lastUpdated: LocalDateTime,
    val products: List<PurchaseProductResponseDTO>,
    val isCompleted: Boolean,
    val totalSpent: Double,
    val totalSaved: Double
)